### PR TITLE
fix assert during compact IDs pass

### PR DESCRIPTION
**Pull request by pow2clk**
During the compact IDs optimization pass, the result IDs of some
basic blocks can change. In spite of this, GetPreservedAnalyses
indicated that the CFG was preserved. But the CFG relies on
the basic blocks having the same IDs. Simply removing this flag
resolves the issue by preventing the CFG check.

However, the CFG retrieval would update the CFG if this bit
is set. So it's not necessary to skip the CFG test even when it
has been changed. So I've removed that early exit too

Addresses https://github.com/KhronosGroup/SPIRV-Tools/issues/1648

### CODE REVIEW

File: source/opt/ir_context.cpp [hunk 1]
```diff
@@ -601,9 +601,6 @@ opt::PostDominatorAnalysis* IRContext::GetPostDominatorAnalysis(
 
 bool ir::IRContext::CheckCFG() {
   std::unordered_map<uint32_t, std::vector<uint32_t>> real_preds;
-  if (!AreAnalysesValid(kAnalysisCFG)) {
-    return true;
-  }
```
[the remark this answers is not in the document]
Comment (s-perron) [reply source/opt/ir_context.cpp]: We don't want to rebuild at this point.  This is just to make sure that if the cfg if valid that it matches reality.  We should put back in.

File: source/opt/compact_ids_pass.h [hunk 2]
```diff
@@ -31,7 +31,7 @@ class CompactIdsPass : public Pass {
   // Return the mask of preserved Analyses.
   ir::IRContext::Analysis GetPreservedAnalyses() override {
     return ir::IRContext::kAnalysisInstrToBlockMapping |
-           ir::IRContext::kAnalysisCombinators | ir::IRContext::kAnalysisCFG |
+           ir::IRContext::kAnalysisCombinators |
```
Comment (s-perron) [inline source/opt/compact_ids_pass.h]: Doing at careful look, we should in Combinators and NameMap as well.

**Comment by s-perron** [comment]
I'll make a suggestion on how to do the test tomorrow.  You can build the analyis using a call in the ir context before running the pass.

**Comment by pow2clk** [comment]
Heh. seems @s-perron ended up contributing more than I did anyway. Thanks for humoring me just the same. :smiley: I'll look into testing as well. I would like to get this in ASAP though since it's affecting testing of dxc. So maybe we can add the test separately.

**Pull request merged**